### ISSUE TITLE
V 8.1.4 causes crash on iOS and inability for hot restart

### ISSUE THREAD

**Issue by comatory** [Baseflow/flutter-permission-handler#639]
## 🐛 Bug Report

I have recently bumped up version of the plugin from `8.1.1` to `8.1.4`. I am getting strange behaviour.
I am asking for location permissions when my app starts up which is working fine. I have another view where I'm asking permissions too. I'm using BLoC (`flutter_bloc`) and after `initState` I call bloc that does `await Permission.location.request();` inside.

When this view is visited, the application crashes. I am attaching `crash.log` that is generated by verbose output of `flutter run` command. I had to sanitize it (remove my username and project name), including application logs.

So it seems it does not like when I ask twice for permissions.

When this crash happens, I am unable to start the application at all, I have to uninstall the app. You can see output of this in `start_crash.log`. It ends with `Lost connection to device [...] DevFS: Deleting filesystem on the device [...] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed`

Lastly, doing hot restart also results with device being disconnected (same error as described above). The crash for hot restart happens after permissions were asked. This problem only occurs on iOS (both physical device and emulator), Android seems to be fine.

Downgrading to `8.1.3` fixes the problem.

**logs**

[crash.log](https://github.com/Baseflow/flutter-permission-handler/files/6877244/crash.log)
[start_crash.log](https://github.com/Baseflow/flutter-permission-handler/files/6877245/start_crash.log)

### Expected behavior

No crash during runtime.

### Reproduction steps

Can't provide unfortunately since I'm working on closed source code. I assume install v `8.1.4`.

### Configuration

**Version:** 8.1.4

**Platform:** 
- [X] :iphone: iOS
- [ ] :robot: Android

** Flutter doctor **

```
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-x64, locale en-CZ)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] IntelliJ IDEA Community Edition (version 2020.2.1)
[✓] VS Code (version 1.58.2)
[✓] Connected device (3 available)
```

**Comment by mvanbeusekom** [comment]
Hi @comatory,

We have resolved a bug last Friday in version 8.1.4 which had similar behavior as you describe (crashes after second time requesting permission). The bug fix was released with version 8.1.4+1 would you mind giving that a try and let us know if this soles your problem as wel?

**Comment by comatory** [comment]
@mvanbeusekom Hi. Thanks for answer. Yes `8.1.4+1` solves this problem.
